src/som/interpreter/actors/Actor.java

```diff
@@ -190,12 +203,73 @@ protected void appendToMailbox(final EventualMessage msg) {
     mailboxExtension.append(msg);
   }
 
-  protected static void handleBreakPoints(final EventualMessage msg, final WebDebugger dbg) {
-    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED && msg.isMessageReceiverBreakpointSet()) {
+  protected static void handleBreakpointsAndStepping(final EventualMessage msg, final WebDebugger dbg, final Actor actor) {
+    if (!VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+      return;
+    }
+
+    synchronized (actor) {
+
+      if (msg.isStepToReceiver() || msg.isMessageReceiverBreakpointSet() || actor.isStepToNextTurn()) {
+        dbg.prepareSteppingUntilNextRootNode();
```

Comment:
why do you need two different things for `msg.isStepToReceiver()` and `msg.isMessageReceiverBreakpointSet()`? Isn't that really the same?